client/js/components/App.react.jsx

```diff
@@ -79,13 +79,20 @@ export default class App extends React.Component {
         this.setState({allParks: parksWithDistance});
     }
 
-    applyFilters(filter) {
+    applyFilters(filters) {
         var visibleParks = this.state.allParks.filter((park) => {
             // FIXME: Convert park ids to numbers when we generate the data
-            var matchingAmenity = !!this.state.amenityLookup && this.state.amenityLookup[filter] && this.state.amenityLookup[filter].map(Number).indexOf(park.park_id) !== -1;
-            var matchingFacility = !!this.state.facilityLookup && this.state.facilityLookup[filter] && this.state.facilityLookup[filter].map(Number).indexOf(park.park_id) !== -1;
-
-            return matchingAmenity || matchingFacility;
+            return filters.reduce( (matches, filter) => {
+                const matchingFacility =
```

Comment:
im a little confused by const here as it looks to me like this value would change fairly often. 

I guess its not really changing though its really a completely new value.